src/plugins/moderation/clear.command.ts

```diff
@@ -0,0 +1,64 @@
+import {ChannelType, ChatInputCommandInteraction, PermissionFlagsBits, TextChannel, NewsChannel, ThreadChannel, userMention, channelMention} from "discord.js";
+import {SlashCommandBuilder, SlashCommandScope,} from "../../builders/SlashCommandBuilder";
+import {useChatCommand} from "../../hooks/useChatCommand";
+
+type GuildTextBasedChannel = TextChannel | NewsChannel | ThreadChannel
+
+const builder = new SlashCommandBuilder()
+    .setName("clear")
+    .setDescription("Bulk deletes messages from a given user/channel.")
+    .addIntegerOption((option) => 
+        option.setName("amount").setDescription("Amount of messages to delete (default 100)").setRequired(false)
+    )
+    .addUserOption((option) =>
+        option.setName("user").setDescription("User whose messages you want to purge.").setRequired(false)
+    )
+    .addChannelOption((option) => 
+        option.setName("channel").setDescription("Channel").setRequired(false).addChannelTypes(ChannelType.GuildText)
+    )
+    .setDefaultMemberPermissions(PermissionFlagsBits.ManageMessages)
+    .setScope(SlashCommandScope.GLOBAL);
+
+    useChatCommand(builder, async (interaction: ChatInputCommandInteraction) => {
+        const user = interaction.options.getUser("user", false);
+        const channel = interaction.options.getChannel("channel", false) as TextChannel;
+        const amount = interaction.options.getInteger("amount", false) || 100;
```

Comment:
In this case, amount will default to 100 if amount is falsey meaning if the user specifies 0 messages, it will actually clear 100. Probably better to use the nullish coalescing operator ??, which will only fallback to 100 if the .getInteger returns null or undefined as opposed to any falsey value.